src/platform/site-wide/announcements/config/index.js

```diff
@@ -1,14 +1,49 @@
+// Node modules.
+import moment from 'moment';
 // Relative imports.
+import Downtime from '../components/Downtime';
 import ExploreVAModal from '../components/ExploreVAModal';
 import FindVABenefitsIntro from '../components/FindVABenefitsIntro';
 import PersonalizationBanner from '../components/PersonalizationBanner';
+import PreDowntime from '../components/PreDowntime';
+import PrePreDowntime from '../components/PrePreDowntime';
 import Profile360Intro from '../components/Profile360Intro';
+import VAMCWelcomeModal, { VAMC_PATHS } from '../components/VAMCWelcomeModal';
 import VAPlusVetsModal from '../components/VAPlusVetsModal';
 import WelcomeToNewVAModal from '../components/WelcomeToNewVAModal';
-import VAMCWelcomeModal, { VAMC_PATHS } from '../components/VAMCWelcomeModal';
+
+// Derive when downtime will start and expire.
+const downtimeStartAtDate = moment('2020-02-29 21:00');
+const downtimeExpiresAtDate = moment('2020-02-29 21:30');
 
 const config = {
   announcements: [
+    {
+      name: 'pre-pre-downtime',
+      paths: /(.)/,
+      component: PrePreDowntime,
+      startsAt: downtimeStartAtDate.clone().subtract(12, 'hours'),
+      expiresAt: downtimeStartAtDate.clone().subtract(1, 'hours'),
+      // The following key-value pairs are just used as props, not in selectors.js.
+      downtimeStartsAt: downtimeStartAtDate.toISOString(),
+      downtimeExpiresAt: downtimeExpiresAtDate.toISOString(),
```

Comment:
I love this, but since this is more complicated than our usual announcements, I wonder if any of this could be abstracted out of the main config. Maybe all three pieces could be moved into a dedicated module where the date arithmetic happens, then we could just apply them to this config using the spread operator?

Reply:
I think I'd want to discuss with you what you're thinking here before I go off trying to implement it 🙂On one hand I like having everything in the same file so I don't have to go back-and-forth, but on the other hand I could see how abstracting it would make it easier to read.

If you could give me more details on what the abstracted module would be called and where it would live, I'd love to give it a go!